openeo/rest/_datacube.py

```diff
@@ -346,13 +349,17 @@ def _ensure_save_result(
 
     if not save_result_nodes:
         # No `save_result` node yet: automatically add it.
-        # TODO: the `save_result` method is not defined on _ProcessGraphAbstraction, but it is on DataCube and VectorCube
-        cube = cube.save_result(format=format or weak_format or default_format, options=options)
-    elif format or options:
-        raise OpenEoClientException(
-            f"{method} with explicit output {'format' if format else 'options'} {format or options!r},"
-            f" but the process graph already has `save_result` node(s)"
-            f" which is ambiguous and should not be combined."
-        )
+        if isinstance(cube, (openeo.DataCube, openeo.VectorCube)):
+            pg_with_save_result = cube.save_result(format=format or weak_format or default_format, options=options)
+        else:
+            raise OpenEoClientException(f"No support to add `save_result` on {cube!r}.")
+    else:
+        if format or options:
```

Comment:
With new SaveResult/StacResource we should actually never get in this branch to throw exception, so can we simplify here?

Reply:
turns out that the new SaveResult/StacResource object allows to cleanly eliminate `_ensure_save_result` as a whole